backends/edk2-compat/edk2-svc-read.c

```diff
@@ -4,241 +4,245 @@
+	} 
+	// if no good files read then count it as a failure
+	if (successCount < 1) {
+		prlog(PR_ERR, "No valid files to print, returning failure\n");
+		return INVALID_FILE;
+	}
 
-// 	return SUCCESS;
-// }
+	return SUCCESS;
+}
 
-// /**
-//  *Does the appropriate read command depending on hrFlag on the file <path>/<var>/data
-//  *@param path , the path to the file with ending '/'
-//  *@param variable , variable name one of {db,dbx,KEK,PK,TS}
-//  *@param hrFlag, 1 for human readable 0 for raw data
-//  *@return SUCCESS or error number
-//  */
-// static int readFileFromSecVar(const char *path, const char *variable, int hrFlag)
-// {
-// 	int extra = 10, rc;
-// 	struct secvar *var = NULL;
-// 	char *fullPath = NULL;
+/**
+ *Does the appropriate read command depending on hrFlag on the file <path>/<var>/data
+ *@param path , the path to the file with ending '/'
+ *@param variable , variable name one of {db,dbx,KEK,PK,TS}
+ *@param hrFlag, 1 for human readable 0 for raw data
+ *@return SUCCESS or error number
+ */
+static int readFileFromSecVar(const char *path, const char *variable, int hrFlag)
+{
+	int extra = 10, rc;
+	struct secvar *var = NULL;
+	char *fullPath = NULL;
 	
-// 	fullPath = malloc(strlen(path) + strlen(variable) + extra);
-// 	if (!fullPath) { 
-// 		prlog(PR_ERR, "ERROR: failed to allocate memory\n");
-// 		return ALLOC_FAIL;
-// 	}
+	fullPath = malloc(strlen(path) + strlen(variable) + extra);
+	if (!fullPath) { 
+		prlog(PR_ERR, "ERROR: failed to allocate memory\n");
+		return ALLOC_FAIL;
+	}
 
-// 	strcpy(fullPath, path);
-// 	strcat(fullPath, variable);
-// 	strcat(fullPath, "/data");
+	strcpy(fullPath, path);
+	strcat(fullPath, variable);
+	strcat(fullPath, "/data");
 
-// 	rc = getSecVar(&var, variable, fullPath);
+	rc = getSecVar(&var, variable, fullPath);
 	
-// 	free(fullPath);
+	free(fullPath);
 
-// 	if (rc) {
-// 		goto out;
-// 	}
-// 	if (hrFlag) {
-// 		if (var->data_size == 0) {
-// 			printf("%s is empty\n", var->key);
-// 			rc = SUCCESS;
-// 		}
-// 		else if (strcmp(var->key, "TS") == 0) 
-// 			rc = readTS(var->data, var->data_size);
-// 		else
-// 			rc = printReadable(var->data, var->data_size, var->key);
+	if (rc) {
+		goto out;
+	}
+	if (hrFlag) {
+		if (var->data_size == 0) {
+			printf("%s is empty\n", var->key);
+			rc = SUCCESS;
+		}
+		else if (strcmp(var->key, "TS") == 0) 
+			rc = readTS(var->data, var->data_size);
+		else
+			rc = printReadable(var->data, var->data_size, var->key);
 
-// 		if (rc)
-// 			prlog(PR_WARNING, "ERROR: Could not parse file, continuing...\n");
-// 	}
-// 	else {
-// 		printRaw(var->data, var->data_size);
-// 		rc = SUCCESS;
-// 	}
+		if (rc)
+			prlog(PR_WARNING, "ERROR: Could not parse file, continuing...\n");
+	}
+	else {
+		printRaw(var->data, var->data_size);
+		rc = SUCCESS;
+	}
 	
-// out:
-// 	dealloc_secvar(var);
+out:
+	dealloc_secvar(var);
 	
 <<<<<<< HEAD
+<<<<<<< HEAD
+=======
+>>>>>>> 83004c7... Port openssl to secvarctl read and write
```

Comment:
I think you might have had some trouble with rebasing: you're got all these conflict markers still.

Reply:
Lesson LEARNED! Always compile and test before force pushing. Yikes.... 
Thanks for pointing this out. I have never had to deal with embedded diff statements but that was an interesting experience. I am surprised I did not see any errors or warnings when cherrypicking the commits. Just when you think you finally understand git. I did get to use git bisect though for the first time to figure out where the diffs first came up, so maybe in some weird way, it was worth it.